portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/sitenavigation/navigationmenu/NavigationMenuWidget.testcase

```diff
@@ -78,6 +78,117 @@ definition {
+		task ("Add a navigation menu in Global site") {
+			NavigationMenusAdmin.openNavigationMenusAdmin(siteURLKey = "global");
+
+			NavigationMenusAdmin.addMenu(menuName = "Global Navigation Menu Name");
+		}
+
+		task ("Add a URL item") {
+			NavigationMenusAdmin.addItem(
+				item = "URL",
+				url = "https://www.liferay.com/",
+				urlName = "Liferay");
+		}
+
+		task ("Assert the URL item was added") {
+			NavigationMenusAdmin.viewItem(itemName = "Liferay");
+		}
+
+		task ("Uncheck the View permission of Guests") {
+			NavigationMenusAdmin.openNavigationMenusAdmin(siteURLKey = "global");
+
+			LexiconEntry.gotoEntryMenuItem(
+				menuItem = "Permissions",
+				rowEntry = "Global Navigation Menu Name");
+
+			IFrame.selectPermissionsFrame();
+
+			AssertChecked(
+				key_roleTitle = "guest",
+				locator1 = "Permissions#CONFIGURATION_PERMISSIONS_VIEW_CHECKBOX");
+
+			Uncheck(
+				key_roleTitle = "guest",
+				locator1 = "Permissions#CONFIGURATION_PERMISSIONS_VIEW_CHECKBOX");
+
+			AssertNotChecked(
+				key_roleTitle = "guest",
+				locator1 = "Permissions#CONFIGURATION_PERMISSIONS_VIEW_CHECKBOX");
+
+			PortletEntry.save();
+		}
+
+		task ("Select Global Navigation Menu in normal site") {
+			JSONLayout.addPublicLayout(
+				groupName = "Test Site Name",
+				layoutName = "Test Page Name");
+
+			Navigator.gotoSitePage(
+				pageName = "Test Page Name",
+				siteName = "Test Site Name");
+
+			NavigationMenusWidget.gotoNavigationMenuWidgetConfig();
+
+			NavigationMenusWidget.chooseMenu(menuName = "Global Navigation Menu Name (Global)");
+
+			Portlet.saveAndCloseConfig();
+
+			AssertVisible(
+				key_pageName = "Liferay",
+				locator1 = "Home#PAGE");
+		}
+
+		task ("Assert Guests can not View the content of Navigation Menu portlet") {
+			User.logoutPG();
+
+			Navigator.openSiteURL(siteName = "Test Site Name");
+
+			AssertElementNotPresent(
+				key_pageName = "Liferay",
+				locator1 = "Home#PAGE");
+		}
+	}
+
+	@description = "This is a test for LPS-107833. Can edit Global Navigation Menu via Navigation Menu widget in normal site."
+	@priority = "4"
+	test EditGlobalNavigationMenuViaWidgetInNormalSite {
+		task ("Add a navigation menu in Global site") {
+			NavigationMenusAdmin.openNavigationMenusAdmin(siteURLKey = "global");
+
+			NavigationMenusAdmin.addMenu(menuName = "Global Navigation Menu Name");
+		}
+
+		task ("Select Global Navigation Menu in normal site") {
+			JSONLayout.addPublicLayout(
+				groupName = "Test Site Name",
+				layoutName = "Test Page Name");
+
+			Navigator.gotoSitePage(
+				pageName = "Test Page Name",
+				siteName = "Test Site Name");
+
+			NavigationMenusWidget.gotoNavigationMenuWidgetConfig();
+
+			NavigationMenusWidget.chooseMenu(menuName = "Global Navigation Menu Name (Global)");
+
+			Portlet.saveAndCloseConfig();
+		}
+
+		task ("Edit Global Navigation Menu via Navigation Menu widget") {
+			NavigationMenusWidget.gotoEditNavigationMenuEntry();
+
+			Pause(locator1 = "5000");
```

Comment:
Can't we use a wait here?

Reply:
We don't need to wait.

Reply:
The pause waits for page loading completely. I'd like to give time to take a screenshot since the following steps is too fast to take photo.

Reply:
ok, makes sense! Thank you for your explanation